jetbrains-rider/src-221-223/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt

```diff
@@ -0,0 +1,97 @@
+
+    override fun expand() = Runnable {
+        val samPanel = getSamPanel()
+        val generator = getSamGenerator()
+        val samSettings = samPanel.getNewProjectSettings()
+
+        val solutionDirectory = getSolutionDirectory()
+            ?: throw Exception(message("sam.init.error.no.solution.basepath"))
+
+        val fileSystem = LocalFileSystem.getInstance()
+        if (!solutionDirectory.exists()) {
+            FileUtil.createDirectory(solutionDirectory)
+        }
+
+        val outDirVf = fileSystem.refreshAndFindFileByIoFile(solutionDirectory)
+            ?: throw Exception(message("sam.init.error.no.virtual.file"))
+
+        val progressManager = ProgressManager.getInstance()
+        val samProjectBuilder = generator.createModuleBuilder()
+        progressManager.runProcessWithProgressSynchronously(
+            {
+                samProjectBuilder.runSamInit(
+                    context.project,
+                    projectNameField.text,
+                    samSettings,
+                    null,
+                    outDirVf
+                )
+            },
+            message("sam.init.generating.template"),
+            false,
+            null
+        )
+
+        // Create solution file
+        val projectFiles =
+            File(solutionDirectory, "src").walk().filter { it.extension == CsprojFileType.defaultExtension } +
+                File(solutionDirectory, "test").walk().filter { it.extension == CsprojFileType.defaultExtension }
+
+        // Get the rest of generated files and copy to "SolutionItems" default folder in project structure
+        val solutionFiles = solutionDirectory.listFiles()?.filter { it.isFile }?.toList() ?: emptyList()
+
+        val solutionFile = ReSharperTemplatesInteraction.createSolution(
+            name = getSolutionName(),
+            directory = solutionDirectory,
+            projectFiles = projectFiles.toList(),
+            protocolHost = context.protocolHost,
+            solutionFiles = solutionFiles
+        ) ?: throw Exception(message("sam.init.error.solution.create.fail"))
+
+        val project = runBlocking {
+            SolutionManager.openExistingSolution(
+                projectToClose = null,
+                forceOpenInNewFrame = false,
+                solutionFile = solutionFile,
+                forceConsiderTrusted = true
+            )
+        } ?: return@Runnable
+
+        vcsPanel?.createInitializer()?.execute(project)
+
+        val modifiableModel = ModuleManager.getInstance(project).modules.firstOrNull()?.rootManager?.modifiableModel ?: return@Runnable
+        try {
+            val progressIndicator = if (progressManager.hasProgressIndicator()) progressManager.progressIndicator else DumbProgressIndicator()
+
+            samProjectBuilder.runPostSamInit(project, modifiableModel, progressIndicator, samSettings, outDirVf)
+        } finally {
+            modifiableModel.dispose()
+        }
+    }
```

Comment:
we might be able to extract more common logic since this is the only part changing, but let's just get this out